lib/cloudant.js

```diff
@@ -368,7 +377,7 @@ Cloudant.prototype.destroy = function destroy(model, id, options, cb) {
         cb(err, result && result.ok ? {count: 1} : {count: 0});
       });
     } else {
-      cb(new Error('could not find matching item in database!'));
+      return cb(null, {count: 0});
```

Comment:
Loopback connectors doesn't return error if item not found in database. It just return the `count` and juggler return error according to the flag `strictDelete` and the `count`, see:
https://github.com/strongloop/loopback-datasource-juggler/blob/3a10c77cbdeb658ffc34e8db95acc7c57f2010c0/lib/dao.js#L2819

Related test cases: https://github.com/strongloop/loopback-datasource-juggler/blob/3a10c77cbdeb658ffc34e8db95acc7c57f2010c0/test/manipulation.test.js#L1837-L1880